src/commonMain/kotlin/exchange.dydx.abacus/processor/markets/MarketProcessor.kt

```diff
@@ -143,9 +146,28 @@ internal class MarketProcessor(parser: ParserProtocol, private val calculateSpar
         output["configs"] = configs(parser.asNativeMap(existing?.get("configs")), payload)
         output["perpetual"] = perpetual(parser.asNativeMap(existing?.get("perpetual")), payload, oraclePrice)
         output.safeSet("line", line(output))
+        output.safeSet("configs.effectiveInitialMarginFraction", calculateEffectiveIMF(output, oraclePrice))
         return calculate(output)
     }
 
+    internal fun calculateEffectiveIMF(output: Map<String, Any>, oraclePrice: Double?): Double {
+        val baseIMF = parser.asDouble(parser.value(output, "configs.initialMarginFraction"))
+        val openInterestUSDC = parser.asDouble(parser.value(output, "perpetual.openInterestUSDC"))
+        val openInterestLowerCap = parser.asDouble(parser.value(output, "perpetual.openInterestLowerCap"))
+        val openInterestUpperCap = parser.asDouble(parser.value(output, "perpetual.openInterestUpperCap"))
+
+        // need nully checks because all properties are optional in the websocket message
+        // clean up after https://linear.app/dydx/issue/OTE-301/audit-websocket-message-types-in-indexer is done
+        if (baseIMF === null) return 1.0
```

Comment:
actually not sure what i should be putting here in case baseIMF does not exist.

Reply:
Maybe effectiveIMF should also be null, if baseIMF is null 🤔

Reply:
hmm... is that going to cause any downstream issues? i was worried if a null `maxLeverage` value would cause abacus or our UI to freak out 
FWIW this should never happen, it just needs to be here because we've typed our socket messages to have all optional attributes

Reply:
nvm just saw that maxleverage can be null. updating accordingly!